man/rgbasm.5

```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
+.Pp
+By default,
+.Nm
+uses the precision specified by the
 .Fl Q
-command-line option.
-You can also specify a precise fixed-point value by appending a
+command-line option and/or by
+.Ic OPT Q
+.Pq see Sx Changing options while assembling ;
+the default precision is 16 bits (meaning the lower 16 bits are used for the fractional part, so the numbers count in 65536ths).
+The precision of a literal can be overridden by appending a
 .Dq q
-to it followed by the number of fractional bits, such as
-.Ql 12.34q8 .
-.Pp
-Since fixed-point values are still just integers, you can use them in normal integer expressions.
-Some integer operators like
-.Sq +
-and
-.Sq -
-don't care whether the operands are integers or fixed-point.
-You can easily truncate a fixed-point number into an integer by shifting it right by 16 bits.
-It follows that you can convert an integer to a fixed-point number by shifting it left.
-.Pp
-The following functions are designed to operate with fixed-point numbers:
+followed by the number of fractional bits:
+.Ql 1234.5q8
+is equal to $0004d280
 .EQ
 delim $$
 .EN
+($= 1234.5 * 2 sup 8$).
+.Pp
+The following functions are designed to operate with fixed-point numbers:
 .Bl -column -offset indent "ATAN2(y, x)"
 .It Sy Name Ta Sy Operation
-.It Fn DIV x y Ta Fixed-point division $( x \[di] y ) \[mu] ( 2 ^ precision )$
-.It Fn MUL x y Ta Fixed-point multiplication $( x \[mu] y ) \[di] ( 2 ^ precision )$
-.It Fn FMOD x y Ta Fixed-point modulo $( x % y ) \[di] ( 2 ^ precision )$
-.It Fn POW x y Ta $x$ to the $y$ power
-.It Fn LOG x y Ta Logarithm of $x$ to the base $y$
+.It Fn DIV x y Ta Fixed-point division
+.It Fn MUL x y Ta Fixed-point multiplication
+.It Fn FMOD x y Ta Fixed-point modulo
+.It Fn POW x y Ta $x sup y$
+.It Fn LOG x y Ta $log sub y ( x )$ (such that
+.Ic LOG Ns ( Ic POW Ns ( x , y ) , y ) == x)
 .It Fn ROUND x Ta Round $x$ to the nearest integer
-.It Fn CEIL x Ta Round $x$ up to an integer
-.It Fn FLOOR x Ta Round $x$ down to an integer
+.It Fn CEIL x Ta Round $x$ up to the nearest integer
+.It Fn FLOOR x Ta Round $x$ down to the nearest integer
 .It Fn SIN x Ta Sine of $x$
 .It Fn COS x Ta Cosine of $x$
 .It Fn TAN x Ta Tangent of $x$
-.It Fn ASIN x Ta Inverse sine of $x$
-.It Fn ACOS x Ta Inverse cosine of $x$
-.It Fn ATAN x Ta Inverse tangent of $x$
+.It Fn ASIN x Ta Inverse sine of $x$ (such that
+.Ic ASIN Ns ( Ic SIN Ns ( x ) ) == x)
+.It Fn ACOS x Ta Inverse cosine of $x$ (such that
+.Ic ACOS Ns ( Ic COS Ns ( x ) ) == x)
+.It Fn ATAN x Ta Inverse tangent of $x$ (such that
+.Ic ATAN Ns ( Ic TAN Ns ( x ) ) == x)
 .It Fn ATAN2 y x Ta Angle between $( x , y )$ and $( 1 , 0 )$
 .El
 .EQ
 delim off
 .EN
 .Pp
-All of these fixed-point functions can take an optional final argument, which is the precision to use.
+Operators not in the list above, such as
+.Sq +
+and
+.Sq - ,
+work correctly with fixed-point operands as well (but
+.Em both
```

Comment:
The example of `1.0 * 2` shows that they don't always both *need* to be fixed-point.

Reply:
That's the hard part to explain: `*` works fine iff *either operand* is *not* fixed-point, whereas `+` works iff *neither* operand is fixed-point. It's a bit of a headache (but that's mathematical properties for you lololol), and thus AFAICT a headache to explain as well.

I wonder if @aaaaaa123456789 has any ideas about this.

Reply:
At its heart, the issue here is how factors combine: this is a result of ab + ac = a(b + c) and ab \* ac = a²(bc). I wonder how much you want to turn this page into a tutorial on fixed point: I fear that an overload of examples would have the wrong effect, teaching people to replicate expressions like cargo cult without understanding _why_ the expressions are behaving like they do. After all, the examples are endless: `1.5q8 * 2.5q8` is `3.75q16`.

Reply:
I just don't think "works fine" is a precise enough explanation. `5.0 * 3 == 5 * 3.0 == 15.0`; that's not some arbitrary property we gave the `*` operator, it's a consequence of how multiplication acts. `5.0 + 1` "works" if you wanted the 1 to be short for 1/Q.

Frankly I'm not sure we even need to point out how `*` acts here; we already provide `MUL`, and we explain that `fixed + fixed` is how you add fixed-point. Advanced users who *want* to combine fixed with integer will soon figure out how that behaves.

Reply:
Then perhaps “give the expected results if used with two fixed-point oeprands”?

Reply:
A lot of this discussion is falling into "I wish this was a different project" territory. (It comes up often: for instance, some people wish there was a GB standard library, but there isn't, so they attempt to stuff unneeded stuff into hardware.inc instead...)
Namely, a lot of this is attempting to be a fixed-point tutorial, which the docs should never even attempt. That would make them terrible (due to size alone, if nothing else) as documentation for users who just want to learn the RGBASM behaviors.

Reply:
Sure, but the goal here is to explain *why* there's no `ADD()` or `SUB()`. (We could add them, honestly.)

Reply:
See the comment below:
> There are no functions for fixed-point addition and subtraction because the '+' and '-' operators already do that when both operands are fixed-point.
Easy-peasy.
You could reword it to:
> There are no functions for fixed-point addition and subtraction,  because the '+' and '-' operators can add and subtract pairs of fixed-point operands.
Or something like that.

Reply:
No please let's *not* add redundant functions just so everything fixed-point can look function-y.

It wouldn't simplify documentation -- it would make the bikeshedding worse, as "ADD(x, y) Fixed-point addition" would be a falsehood, and "ADD(x, y) Fixed-point addition (which is the same as integer addition, but this function is included for ??? purposes)" would be a weirdly bloated documentation of an exceptionally simple fixpoint operation.

(Also `add` and `sub` are opcodes, are we certain there'd be no parsing conflicts?)

Reply:
> You could reword it to:
> 
> > There are no functions for fixed-point addition and subtraction,  because the '+' and '-' operators can add and subtract pairs of fixed-point operands.

Looks good to me!

---

<details><summary>Results of the nerd-snipe:</summary>

(There would be no parsing conflicts, because instructions and expressions are never expected in the same context. *One of the few nice things about a language that doesn't have expression-as-statement?* :stuck_out_tongue:)

</details>